woodwork/datatable.py

```diff
@@ -1013,17 +1013,26 @@ def mutual_information_dict(self, num_bins=10, nrows=None):
             nrows (int): The number of rows to sample for when determining mutual info.
                 If specified, samples the desired number of rows from the data.
                 Defaults to using all rows.
-
+            include_index (bool): If True, the column specified as the index will be
+                included as long as its LogicalType is valid for mutual information calculations.
+                If False, the index column will not have mutual information calculated for it.
+                Defaults to False.
         Returns:
             list(dict): A list containing dictionaries that have keys `column_1`,
             `column_2`, and `mutual_info` that is sorted in decending order by mutual info.
             Mutual information values are between 0 (no mutual information) and 1
             (perfect dependency).
         """
         valid_types = get_valid_mi_types()
+
         valid_columns = [col.name for col in self.columns.values() if (
             col.name != self.index and _get_ltype_class(col.logical_type) in valid_types)]
```

Comment:
Should we remove the check here? `col.name != self.index`

Reply:
If we make sure the index gets included here, then we'll need to remove it later if `include_index` is False or if its logical type is not valid for mutual info, so I guess it'd just be the reverse of the way it currently is. I'm fine with either one

Reply:
I think if we do 1 less check to `_get_ltype_class` and 1 less check to see if its in `valid_types` its better. 
```python
        valid_types = get_valid_mi_types()

        valid_columns = [col.name for col in self.columns.values() if _get_ltype_class(col.logical_type) in valid_types)]

        if not include_index and self.index is not None:
            valid_columns.remove(self.index)
```

Reply:
That makes sense to me. Made this change (using `remove` for dropping column names from a list, though, bc pop is for indices)